extract.py

```diff
@@ -0,0 +1,139 @@
+"""
+Extracts information from scraped websites, 
+to reduce noise in AI model training.
+"""
+import logging
+import copy
+import regex as re
+from bs4 import BeautifulSoup
+
+class NoSoup(Exception):
+    """
+    Data extractor hasn't generated a soup, and cannot start extracting data.
+    """
+    pass
+
+class DataExtractor:
+    """
+    Extracts information from scraped websites.
+    """
+    def __init__(self):
+        self.soup = None
+        self.string_filter_list = [ # Last filter, removes strings from lists.
+            re.compile(r'\d\d\d\d') # Remove strings with years (i.e. a string containing '2013')
+            ]
+
+    def __str__(self):
+        return self.extract()
+
+    def open_file(self, file_path):
+        """
+        Opens a file and creates a soup object from a file.
+        :param file_path: path to an html file.
+        """
+        try:
+            with open(file_path, encoding='utf-8') as fp:
+                self.create_soup_from_html(fp)
+        except (FileNotFoundError):
+            logging.error("Couldn't find %s!", file_path)
+    
+    def create_soup_from_html(self, raw_html):
```

Comment:
This function should be private

Reply:
As discussed, we change the name of the function instead of marking it as private.